common/source/world/Biome.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * =====================================================================================
+ *
+ *  OpenMiner
+ *
+ *  Copyright (C) 2018-2020 Unarelith, Quentin Bazin <openminer@unarelith.net>
+ *  Copyright (C) 2019-2020 the OpenMiner contributors (see CONTRIBUTORS.md)
+ *
+ *  This file is part of OpenMiner.
+ *
+ *  OpenMiner is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  OpenMiner is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with OpenMiner; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * =====================================================================================
+ */
+#ifndef BIOME_HPP_
+#define BIOME_HPP_
+
+#include <string>
+#include <vector>
+
+#include <gk/core/IntTypes.hpp>
+
+#include "FloraPlacementEntry.hpp"
+#include "ISerializable.hpp"
+#include "OrePlacementEntry.hpp"
+#include "TreePlacementEntry.hpp"
+
+class Biome : public ISerializable {
+	public:
+		Biome() = default;
+		Biome(u32 id, const std::string &stringID, const std::string &label);
+
+		void serialize(sf::Packet &packet) const override;
+		void deserialize(sf::Packet &packet) override;
+
+		u16 id() const { return m_id & 0xffff; }
```

Comment:
Why do you store the ID as `u32` and return them as `u16`?

If you saw that in `Block`/`Item`, they work differently because they have runtime data associated to them. :p 

For biome ID, I think a `u16` is enough.